src/main/java/org/jenkinsci/maven/plugins/hpi/RunMojo.java

```diff
@@ -262,28 +260,20 @@ public void execute() throws MojoExecutionException, MojoFailureException {
                 throw new MojoExecutionException("Version of "+a.getId()+" is inconsistent with "+ba.getId());
         }
 
-        // set HUDSON_HOME
-        SystemProperty sp = new SystemProperty();
-        sp.setName("HUDSON_HOME");
-        sp.setValue(jenkinsHome.getAbsolutePath());
-        sp.setIfNotSetAlready();
+        // set JENKINS_HOME
+        setSystemPropertyIfEmpty("HUDSON_HOME",jenkinsHome.getAbsolutePath());
+        setSystemPropertyIfEmpty("JENKINS_HOME",jenkinsHome.getAbsolutePath());
```

Comment:
this does break backwards compatibility with anyone who sets HUDSON_HOME and uses hpi:run - as JENKINS_HOME will take precedence over HUDSON_HOME (IIRC)
I'm inclined to say if you set one you set both - or only set JENKINS_HOME.

But we should be encouraging devs to use JENKINS_HOME as HUDSON is long dead :-)